scot/datatools.py

```diff
@@ -154,17 +168,48 @@ def randomize_phase(data):
         plot(y.T + [1.5, -1.5]), axis([0,1000,-3,3])
         plt.show()
     """
+    rng = check_random_state(random_state)
     data = np.asarray(data)
     data_freq = np.fft.rfft(data)
-    data_freq = np.abs(data_freq) * np.exp(1j*np.random.random_sample(data_freq.shape)*2*np.pi)
+    data_freq = np.abs(data_freq) * np.exp(1j*rng.random_sample(data_freq.shape)*2*np.pi)
     return np.fft.irfft(data_freq, data.shape[-1])
 
 
-def atleast_3d(x):
```

Comment:
I didn't remove it, I just moved it to the top (which isn't necessary of course, but in the course of debugging the cyclic import I tried moving it around).